### ISSUE TITLE
issue during 1st iteration of pretraining

### ISSUE THREAD

**Issue by sungheedong** [facebookresearch/av_hubert#75]
Hi.

I conduct pretraining of 1st iteration.
And I found the following issue during training.

######
 UserWarning: Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance. 
grad.sizes() = [64, 1, 5, 7, 7], strides() = [245, 1, 49, 7, 1] 
bucket_view.sizes() = [64, 1, 5, 7, 7], strides() = [245, 245, 49, 7, 1] (Triggered internally at ../torch/csrc/distributed/c10d/reducer.cpp:325.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass  

####

What is the problem?

And, how much time did you take to pretrain AV hubert model in the first iteration?
I think it takes too long in my system... I'm using 6 GPU (32 GB, A100) for 1st iteration of pretraining.
Can you tell us the spec of GPU system used for pretraining? and it's training time?

Thank you so much.

**Comment by chevalierNoir** [comment]
Hi,

We don't have this warning during training. This is usually caused by the non-contiguous shape by transpose operation. Hope [the related issue thread](https://github.com/pytorch/pytorch/issues/47163) can help. 

In terms of speed, one gradient update takes about 0.6s on 32 V100 GPUs, which means ~35 hours for one iteration of pre-training (400K update). If you use smaller number of GPUs, you'd better increase `optimization.update_freq` to match the effective batch size to 32-GPU setting. Otherwise, the model performance may be impacted. You can find more details on [this thread](https://github.com/facebookresearch/av_hubert/issues/66#issuecomment-1284867011).
